components/director/internal/destinationfetchersvc/service.go

```diff
@@ -0,0 +1,352 @@
+package destinationfetchersvc
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/pkg/apperrors"
+	"github.com/kyma-incubator/compass/components/director/pkg/config"
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+	"github.com/kyma-incubator/compass/components/director/pkg/persistence"
+	"github.com/kyma-incubator/compass/components/director/pkg/resource"
+	"golang.org/x/sync/semaphore"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	correlationIDPrefix = "sap.s4:communicationScenario:"
+	s4HANAType          = "SAP S/4HANA Cloud"
+	regionLabelKey      = "region"
+)
+
+//go:generate mockery --name=UUIDService --output=automock --outpkg=automock --case=underscore --disable-version-string
+// UUIDService missing godoc
+type UUIDService interface {
+	Generate() string
+}
+
+//go:generate mockery --name=DestinationRepo --output=automock --outpkg=automock --case=underscore --disable-version-string
+// DestinationRepo missing godoc
+type DestinationRepo interface {
+	Upsert(ctx context.Context, in model.DestinationInput, id, tenantID, bundleID, revision string) error
+	DeleteOld(ctx context.Context, latestRevision, tenantID string) error
+}
+
+//go:generate mockery --name=LabelRepo --output=automock --outpkg=automock --case=underscore --disable-version-string
+// LabelRepo missing godoc
+type LabelRepo interface {
+	GetSubdomainLabelForSubscribedRuntime(ctx context.Context, tenantID string) (*model.Label, error)
+	GetByKey(ctx context.Context, tenant string, objectType model.LabelableObject, objectID, key string) (*model.Label, error)
+}
+
+//go:generate mockery --name=BundleRepo --output=automock --outpkg=automock --case=underscore --disable-version-string
+// BundleRepo missing godoc
+type BundleRepo interface {
+	ListByDestination(ctx context.Context, tenantID string, destination model.DestinationInput) ([]*model.Bundle, error)
+}
+
+//go:generate mockery --name=TenantRepo --output=automock --outpkg=automock --case=underscore --disable-version-string
+// TenantRepo missing godoc
+type TenantRepo interface {
+	ListBySubscribedRuntimes(ctx context.Context) ([]*model.BusinessTenantMapping, error)
+}
+
+// DestinationService missing godoc
+type DestinationService struct {
+	Transactioner      persistence.Transactioner
+	UUIDSvc            UUIDService
+	Repo               DestinationRepo
```

Comment:
Can you rename the variable to `DestRepo` or something that implies that it's a destination service repo. It's just that that's how we've done it thoughtout all service structs and I'd like to keep it consistent